core/coreapi/unixfs.go

```diff
@@ -96,7 +97,29 @@ func (api *UnixfsAPI) Add(ctx context.Context, files files.Node, opts ...options
 	bserv := blockservice.New(addblockstore, exch) // hash security 001
 	dserv := dag.NewDAGService(bserv)
 
-	fileAdder, err := coreunix.NewAdder(ctx, pinning, addblockstore, dserv)
+	// add a sync call to the DagService
+	// this ensures that data written to the DagService is persisted to the underlying datastore
+	// TODO: propagate the Sync function from the datastore through the blockstore, blockservice and dagservice
+	var syncDserv *syncDagService
+	if settings.OnlyHash {
+		syncDserv = &syncDagService{
+			DAGService: dserv,
+			syncFn:     func(c cid.Cid) error { return nil },
+		}
+	} else {
+		syncDserv = &syncDagService{
+			DAGService: dserv,
+			syncFn: func(c cid.Cid) error {
+				ds := api.repo.Datastore()
+				if err := ds.Sync(bstore.BlockPrefix.ChildString(c.String())); err != nil {
```

Comment:
That's not what block keys look like. Take a look at `go-ipfs-ds-help.CidToDsKey`.

Reply:
Yep, thanks

Reply:
i.e. any reason to not just sync the whole Blockstore?